### ISSUE TITLE
In Falco 0.31.1, the triggered rules are not written to the log file

### ISSUE THREAD

**Issue by pmusa** [falcosecurity/falco#1957]
**Describe the bug**

(cc: @jasondellaluce )

When installing Falco 0.31.1 (which is currently the latest and the default) using Docker or Helm in Ubuntu 20.04 (did not test others), rules do not show up in the logs.

**How to reproduce it**

Start a ubuntu machine and run the following commands:

```
apt-get update
apt-get install -y docker.io
docker pull falcosecurity/falco-driver-loader:0.31.1
docker run --rm -i -t \
  --privileged \
  -v /root/.falco:/root/.falco \
  -v /proc:/host/proc:ro \
  -v /boot:/host/boot:ro \
  -v /lib/modules:/host/lib/modules:ro \
  -v /usr:/host/usr:ro \
  -v /etc:/host/etc:ro \
  falcosecurity/falco-driver-loader:0.31.1 && echo "Falco drivers installed!"
docker pull falcosecurity/falco-no-driver:0.31.1
docker run --rm -d \
  --privileged \
  --name "falco_training" \
  --security-opt apparmor:unconfined \
  -p 8765:8765 \
  -e HOST_ROOT=/ \
  --cap-add SYS_PTRACE \
  --pid=host $(ls /dev/falco* | xargs -I {} echo --device {}) \
  -v /var/run/docker.sock:/var/run/docker.sock \
  falcosecurity/falco-no-driver:0.31.1
docker logs falco_training
find /root -name "id_rsa"
docker logs falco_training # nothing in the logs
```

If instead of `0.31.1`, you use `0.31.0` the rule shows up in the logs.

The same happens if you are using helm (v1.17.4 is currently the latest and default).

```
helm repo add falcosecurity https://falcosecurity.github.io/charts
helm repo update
helm install falco --version 1.17.4 falcosecurity/falco
k logs <podname>
find /root -name "id_rsa"
k logs <podname> # nothing in the logs
```

If you use `1.17.3` it works.

**Comment by jasondellaluce** [comment]
Hi @pmusa, after some investigation I think this is related to the Docker logs buffering.

One indicator of this hypothesis is that events are printed to the logs after a bunch of alerts are triggered, and they state the right timestamp. The second point is that, as for my testing, this happens only when the container is launched as detatched. To mitigate this, I tried launching the container like this and it worked:
```
docker run --rm -d \
  --privileged \
  --name "falco_training" \
  --security-opt apparmor:unconfined \
  -p 8765:8765 \
  -e HOST_ROOT=/ \
  --cap-add SYS_PTRACE \
  --pid=host $(ls /dev/falco* | xargs -I {} echo --device {}) \
  -v /var/run/docker.sock:/var/run/docker.sock \
  falcosecurity/falco-no-driver:0.31.1 \
  /usr/bin/falco -o file_output.enabled=true -o file_output.filename=/dev/stdout
```
I think this example works for me because it bypasses the docker buffering someway. I think similar spot-checks can be performed by testing the http/grpc/syslog outputs. Can you double check?

**Comment by LucaGuerra** [comment]
After a bit of investigation I believe this is a known issue which is not really related to Falco and not new to 0.31.1 .
I believe that if you need to run Falco in such a way that events appear immediately in the log file you can add `--tty` to the Docker command line (or `tty: true` if running in kubernetes or compose). Can you try that?

**Comment by LucaGuerra** [comment]
Thanks a lot @pmusa !

The reason why I believe that this behavior is not new in 0.31.1 is not because I released 0.31.1 😬 but because I found these:
* https://github.com/falcosecurity/falco/issues/1198
* https://github.com/falcosecurity/falco/issues/1291
* https://github.com/falcosecurity/falco/pull/1296

I believe we can close this issue. In addition, I added a note about this to the documentation here: https://github.com/falcosecurity/falco-website/pull/578
